tests/legacy-cli/e2e/utils/process.ts

```diff
@@ -34,16 +35,18 @@ function _exec(options: ExecOptions, cmd: string, args: string[]): Promise<Proce
 
   const cwd = options.cwd ?? process.cwd();
   const env = options.env ?? process.env;
+  const windowsMode = isWindowsTestMode();
 
   console.log(
     `==========================================================================================`,
   );
 
   // Ensure the custom npm and yarn global bin is on the PATH
-  // https://docs.npmjs.com/cli/v8/configuring-npm/folders#executables
+  // https://docs.npmjs.com/cli/v8/configuring-npm/folders#executables.
+  // On Linux, macOS platforms, `bin/` is needed.
   const paths = [
-    join(getGlobalVariable('yarn-global'), 'bin'),
-    join(getGlobalVariable('npm-global'), process.platform.startsWith('win') ? '' : 'bin'),
+    join(getGlobalVariable('yarn-global'), windowsMode !== null ? '' : 'bin'),
+    join(getGlobalVariable('npm-global'), windowsMode !== null ? '' : 'bin'),
```

Comment:
```suggestion
    join(getGlobalVariable('yarn-global'), windowsMode === null ? 'bin' : ''),
    join(getGlobalVariable('npm-global'), windowsMode === null ? 'bin' : ''),
```

I don't know why but I felt like I should suggest that we do equality check instead of inequality check.

Reply:
It's funny because somehow my brain prefers the other way :D